Autoborger QoL Changes

N3X15 opened the pull request:
# Executive Summary
Autoborging QoL changes for both borgs and AI.  Chiefly, borgs don't get namepick dialogs in their faces upon autoborging, and AI have a few new options.

# Justification
Seeing one too many cyborgs named "HELP AI MALF" or "AUTOBORGER IN CORE"

# Screenshots
![2017-02-13_13-07-25](https://cloud.githubusercontent.com/assets/110073/22903430/662eaf1c-f1ed-11e6-98a5-d893d548f980.png)
![dreamseeker_2017-02-13_12-38-06](https://cloud.githubusercontent.com/assets/110073/22903434/6e2f6db4-f1ed-11e6-8653-5842a278c40a.png)
![dreamseeker_2017-02-13_12-44-50](https://cloud.githubusercontent.com/assets/110073/22903439/760e432a-f1ed-11e6-95d2-dcffba6d17fd.png)


# Testing
![dreamseeker_2017-02-13_13-02-22](https://cloud.githubusercontent.com/assets/110073/22903203/b9d5743a-f1ec-11e6-879f-0810e387d087.png)


1. Spawned autoborger on tiny, cooldown_duration=0
2. Spawned in as human
3. Rested on belt
 * Name default cyborg-###, namepick allowed.
4. Transmog'd to human
5. VV'd forced_module to Peacekeeper
6. Rested on belt
 * Name Peacekeeper cyborg-###, module forced to peacekeeper, namepick disallowed.
7. Transmogged to AI
8. Changed enabled namepick to OFF
9. Transmogged to human
10. Rested on belt.
 * Became peacekeeper
 * Named peacekeeper cyborg-###
 * Namepick disabled and gave the correct message.
11. Transmogged to AI
12. Changed name type to SILLY
13. Transmogged to human
14. Rested on belt.
 * Became peacekeeper
 * Named Robocop-###
 * Namepick disabled and gave the correct message.

# Changelog
:cl:
* rscadd: Autoborged humans no longer get name picking dialogue after being autoborged. Namepick verb will still work, if the AI chooses to.
* rscadd: AI can select between normal and silly default names in the autoborger.
* rscadd: AI can choose to disable namepick.
* rscadd: Autoborgs get a notification about being allowed to change their name or not, and how to do so.

mrpain666 commented:
If this has happened recently, let me know about it and I'll take care of it. It's a dick move on their part towards the AI and they deserve to get their shit slapped for it.

9600bauds commented:
Seems like an administration issue, not a code issue

N3X15 commented:
@mrpain666 Not entirely sure it was on purpose, as it could be someone typing a warning and not paying attention to what field they were focused on.

CptWad commented:
Id prefer a list of robot names instead be chosen

Sprok0 commented:
I'd more like a solution to new autoborgs going braindead and clogging up the output end of the borger

Aurx commented:
This is just as much of an issue, basically nobody uses default names.  
You'll convert the issue from "shitters can be shit and out the malf" to "every autoborged borg's name outs the malf".  

This is also 100% administrative issue, we can just ban the occasional fuckwit.

Shadowmech88 commented:
Default borg names are terrible.
This should use a pre-defined list of decent borg names, or maybe even just pick from the random AI names list.

Exxion commented:
What Aurx said
Seeing borgs named anything on that list would out the AI nearly as quickly as "AI MALF", except now it would be MANDATORY

SonixApache commented:
can you make it so it gibs people that are silicon/antagbanned instead

SonixApache commented:
Autoborger controls for AI would be a big plus, default names with a single-use namepick verb would be perfect. No more annoying name prompt while you're typing shit - just get straight to the action.

Lokiamis commented:
@N3X15 
>Not entirely sure it was on purpose, as it could be someone typing a warning and not paying attention to what field they were focused on.
Then add an "are you sure?" prompt like when cyborgs are choosing their icon. Or don't, because I've never seen this happen before.

N3X15 commented:
> Okay I've reworked the autoborger name forcing logic again.
> Once autoborged, you do not get a name-picking dialogue
> HOWEVER, AI can let you set it later via Namepick, changed with a toggle on the autoborg menu
> AI can also switch between normal names, and the silly names as default
> so even if you get a silly name, if the AI has set the namepicking enable, you can still change it
> The autoborg silly name list is a text file in the config/names directory like all the others, except it allows replacing {AINAME} with the AI's name
Testing the new behaviour, then I'll remove the DNM.

N3X15 commented:
Testing complete.

SonixApache commented:
hi my name is sonix and +Outbreak Handling Unit {###} is my favorite line in this PR

Blithering commented:
Requesting 'Damage Control' as a borgo name.
